Client/Application/AppDelegate.swift

```diff
@@ -355,7 +356,12 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UIViewControllerRestorati
 
         // Cleanup can be a heavy operation, take it out of the startup path. Instead check after a few seconds.
         DispatchQueue.main.asyncAfter(deadline: .now() + 5.0) {
-            self.profile?.cleanupHistoryIfNeeded()
+            if let profile = self.profile {
+                profile.cleanupHistoryIfNeeded()
+                // If we recorded any Sync ping events to send for received
+                // tabs, send them now.
+                profile.sendQueuedSyncEvents()
```

Comment:
Isn't this performed as part of sync telemetry?
Maybe when sync runs and sees it has no sync work to do it doesn't send any telemetry thus never does this?

Reply:
It is—I was wondering if we might ever get into a case where we receive a tab, but don't sync...when Sync has no work to do, or if `BrowserSyncManager#endSyncing` gets called when we're not running the main app (but then the telemetry events should just get queued up until the next time the app _is_ launched, right?), and iOS doesn't have Sync decoupling like Desktop. Maybe it's safe to get rid of this and rely on `endSyncing` submitting events, knowing we'll get them eventually, if not right after receiving the tab?

Reply:
> It is—I was wondering if we might ever get into a case where we receive a tab, but don't sync

Yes to that, could happen, and more sendtab events get queued up until they are sent, but that is ok.
ASIDE: that queue is reliable because it uses the special (and cool) system of queuing the sendtab telemetry in the app prefs (which iOS has guarantees about, i.e. it has good async read/write guarantees). 

> `BrowserSyncManager#endSyncing` gets called when we're not running the main app (but then the telemetry events should just get queued up until the next time the app _is_ launched, right?)

agreed, the queue of events just keeps building until the end of a normal sync and they get flushed.

> Maybe it's safe to get rid of this and rely on `endSyncing` submitting events, knowing we'll get them eventually, if not right after receiving the tab?

+1, they will get sent eventually, I prefer just that one location, seems easier to reason about from my perspective.